tests/core/test_context.py

```diff
@@ -285,6 +287,32 @@ def test_plan_execution_time():
     )
 
 
+def test_python_model_empty_df_raises(sushi_context, capsys):
+    @model(
+        "memory.sushi.test_model",
+        columns={"col": "int"},
+        kind=dict(name=ModelKindName.FULL),
+    )
+    def entrypoint(context, **kwargs):
+        yield pd.DataFrame({"col": []})
+
+    test_model = model.get_registry()["memory.sushi.test_model"].model(
+        module_path=Path("."), path=Path(".")
+    )
+    sushi_context.upsert_model(test_model)
+
+    capsys.readouterr()
+    with pytest.raises(SQLMeshError):
+        sushi_context.plan(no_prompts=True, auto_apply=True)
+
+    assert (
+        "Cannot construct source query from an empty \nDataFrame. This error "
+        "is commonly related to Python models that produce no data.\nFor such "
+        "models, consider yielding from an empty generator if the resulting set "
+        "\nis empty, i.e. use `yield from ()`"
+    ) in capsys.readouterr().out
```

Comment:
I tested this using `capsys` because if I do `pytest.raises(SQLMeshError, match=...)` it captures the `Plan application failed.` exception which [doesn't include](https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/plan/evaluator.py#L174-L175) the exception of interest in its context.